internal/xds/translator/authorization.go

```diff
@@ -0,0 +1,265 @@
+		// filter is added in a route.
+		return fmt.Errorf("route already contains rbac config: %+v", route)
+	}
+
+	var (
+		authorization = irRoute.Security.Authorization
+		allowAction   *anypb.Any
+		denyAction    *anypb.Any
+		sourceIPInput *anypb.Any
+		ipMatcher     *anypb.Any
+		matcherList   []*matcherv3.Matcher_MatcherList_FieldMatcher
+		err           error
+	)
+
+	allow := &rbacconfigv3.Action{
+		Name:   "ALLOW",
+		Action: rbacconfigv3.RBAC_ALLOW,
+	}
+	if allowAction, err = anypb.New(allow); err != nil {
+		return err
+	}
+
+	deny := &rbacconfigv3.Action{
+		Name:   "DENY",
+		Action: rbacconfigv3.RBAC_DENY,
+	}
+	if denyAction, err = anypb.New(deny); err != nil {
+		return err
+	}
+
+	// Build a list of matchers based on the rules.
+	// The matchers will be evaluated in order, and the first one that matches
+	// will be used to determine the action, the rest of the matchers will be
+	// skipped.
+	// If no matcher matches, the default action will be used.
+	for _, rule := range authorization.Rules {
+		// Build the IPMatcher based on the client CIDRs.
+		ipRangeMatcher := &ipmatcherv3.Ip{
+			StatPrefix: "source_ip",
+		}
+
+		for _, cidr := range rule.Principal.ClientCIDRs {
+			ipRangeMatcher.CidrRanges = append(ipRangeMatcher.CidrRanges, &configv3.CidrRange{
+				AddressPrefix: cidr.IP,
+				PrefixLen: &wrapperspb.UInt32Value{
+					Value: cidr.MaskLen,
+				},
+			})
+		}
+
+		if ipMatcher, err = anypb.New(ipRangeMatcher); err != nil {
+			return err
+		}
+
+		if sourceIPInput, err = anypb.New(&networkinput.SourceIPInput{}); err != nil {
+			return err
+		}
+
+		// Determine the action for the current rule.
+		ruleAction := allowAction
+		if rule.Action == egv1a1.AuthorizationActionDeny {
+			ruleAction = denyAction
+		}
+
+		// Add the matcher generated with the current rule to the matcher list.
+		matcherList = append(matcherList, &matcherv3.Matcher_MatcherList_FieldMatcher{
+			Predicate: &matcherv3.Matcher_MatcherList_Predicate{
+				MatchType: &matcherv3.Matcher_MatcherList_Predicate_SinglePredicate_{
+					SinglePredicate: &matcherv3.Matcher_MatcherList_Predicate_SinglePredicate{
+						Input: &cncfv3.TypedExtensionConfig{
+							Name:        "source_ip",
+							TypedConfig: sourceIPInput,
+						},
+						Matcher: &matcherv3.Matcher_MatcherList_Predicate_SinglePredicate_CustomMatch{
+							CustomMatch: &cncfv3.TypedExtensionConfig{
+								Name:        "ip_matcher",
+								TypedConfig: ipMatcher,
+							},
+						},
+					},
+				},
+			},
+			OnMatch: &matcherv3.Matcher_OnMatch{
+				OnMatch: &matcherv3.Matcher_OnMatch_Action{
+					Action: &cncfv3.TypedExtensionConfig{
+						Name:        "action",
+						TypedConfig: ruleAction,
+					},
+				},
+			},
+		})
+	}
+
+	// Set the default action.
+	defaultAction := denyAction
+	if authorization.DefaultAction == egv1a1.AuthorizationActionAllow {
+		defaultAction = allowAction
+	}
+
+	routeCfgProto := &rbacv3.RBACPerRoute{
```

Comment:
can we represent this as `matcherList` where we append the default action as a matcher to the end of the `mactherList` ?